project/src/main/webapp/shared.js

```diff
@@ -55,31 +82,124 @@ async function moveTagsFromPoolToInput(clickedAdder, tagId) {
-      if ($(tagId).prevAll().length > 2) {
-        const otherId = tagId === '#interests' ? '#skills' : '#interests';
-        $(otherId).prev().remove();
-        $(tagId).prev().prev().remove();
-      }
-    });
-    $(tagId).on('itemRemoved', function(event) {
-      addTagBackToPool(event.item, tagId);
-    });
+  $(tagId).on('itemAdded', function() {
+    if ($(tagId).prevAll().length > 2) {
+      const otherId = tagId === '#interests' ? '#skills' : '#interests';
+      $(otherId).prev().remove();
+      $(tagId).prev().prev().remove();
+    }
+  });
+  $(tagId).on('itemRemoved', function(event) {
+    addTagBackToPool(event.item, tagId);
+  });
+  getTagsScriptWithCallback(function() {
     $(tagId).tagsinput('refresh');
     $(tagId).tagsinput('add', $(clickedAdder).next().html());
   });
 }
 
+function getTagsScriptWithCallback(fn) {
+  $.getScript(
+      'https://cdnjs.cloudflare.com/ajax/libs/bootstrap-tagsinput/0.8.0/bootstrap-tagsinput.js',
+      fn);
+}
+
 /** Puts item back into pool if it was a preset */
 async function addTagBackToPool(item, tagId) {
   const prefilledItems = await getPrefilledInformation(tagId);
   if (prefilledItems.includes(item)) {
-    $(tagId + '-div > .prefilled-pool > .btn-group:contains("' + item + '")').show();
+    $(tagId + '-div > .prefilled-pool > .btn-group:contains("' + item + '")')
+        .show();
   }
 }
 
 /** Remove tag from pool of options once it is chosen */
 function removeTagsFromPool(clickedAdder) {
   $(clickedAdder).parent().hide();
 }
+
+function togglePrefilledSkills() {
+  $('#prefilled-skills').toggle();
+}
+
+function togglePrefilledInterests() {
+  $('#prefilled-interests').toggle();
+}
+
+/** Writes out relevant details to an event card */
+function populateEventContainer(event, containerId) {
+  const indexOfEventCard = 25;
+  $.get('event-card.html', function(eventCardTotal) {
+    const eventCardId = `event-${event.eventId}`;
+    const eventCard = $(eventCardTotal).get(indexOfEventCard);
+    $(eventCard).attr('id', eventCardId);
+    $(`#${containerId}`).append(eventCard);
+    $(`#${eventCardId} #event-card-title`).html(event.name);
+    $(`#${eventCardId} #event-card-description`).html(event.description);
+    $(`#${eventCardId} #event-card-date`)
+        .html(
+            buildDate(
+                event.date.year, event.date.month, event.date.dayOfMonth));
+    $(`#${eventCardId} #event-card-time`).html(event.time);
+    $(`#${eventCardId} #event-card-location`).html(event.location);
+    $(`#${eventCardId} #event-card-volunteers`)
+        .html(buildVolunteers(event.opportunities));
+    buildAsLabels(
+        `#${eventCardId} #event-card-labels`, event.labels, 'interests');
+    buildSkillsAsLabels(
+        `#${eventCardId} #event-card-labels`, event.opportunities);
+    addLinkToRegister(eventCardId);
+    addLinkToDetails(eventCardId);
+  });
+}
+
+/** Adds a hyperlink to the registration button of event card */
```

Comment:
'Adds a hyperlink to register  to the registration button of event card ' or some comment to differentiate why &register=true vs false in addLinkToRegister() on line 157

Reply:
I'd like to keep the JSDoc this way so that it doesn't leak details about implementation